bundles/org.openhab.binding.wlanthermo/src/main/java/org/openhab/binding/wlanthermo/internal/api/nano/WlanThermoNanoV1CommandHandler.java

```diff
@@ -0,0 +1,275 @@
+                            Color c = Color.decode(color);
+                            return HSBType.fromRGB(c.getRed(), c.getGreen(), c.getBlue());
+                        } else {
+                            return UnDefType.UNDEF;
+                        }
+                    case CHANNEL_COLOR_NAME:
+                        String colorHex = channel.getColor();
+                        if (colorHex != null && !colorHex.isEmpty()) {
+                            return new StringType(WlanThermoNanoV1Util.toColorName(colorHex));
+                        } else {
+                            return UnDefType.UNDEF;
+                        }
+                }
+            }
+        } else if (channelUID.getId().startsWith(CHANNEL_PITMASTER_1)) {
+            if (data.getPitmaster() != null && data.getPitmaster().getPm() != null
+                    && data.getPitmaster().getPm().size() > 0) {
+                Pm pm = data.getPitmaster().getPm().get(0);
+                switch (channelUID.getIdWithoutGroup()) {
+                    case CHANNEL_PITMASTER_CHANNEL_ID:
+                        return new DecimalType(pm.getChannel());
+                    case CHANNEL_PITMASTER_PIDPROFILE:
+                        return new DecimalType(pm.getPid());
+                    case CHANNEL_PITMASTER_DUTY_CYCLE:
+                        return new DecimalType(pm.getValue());
+                    case CHANNEL_PITMASTER_SETPOINT:
+                        return new QuantityType<>(pm.getSet(), unit);
+                    case CHANNEL_PITMASTER_STATE:
+                        return new StringType(pm.getTyp());
+                }
+            } else {
+                return UnDefType.UNDEF;
+            }
+        }
+        throw new WlanThermoUnknownChannelException(channelUID);
+    }
+
+    public static boolean setState(ChannelUID channelUID, Command command, Data data) {
+        String groupId;
+        try {
+            groupId = requireNonNull(channelUID.getGroupId());
+        } catch (WlanThermoInputException e) {
+            return false;
+        }
+
+        List<Channel> channelList = data.getChannel();
+        System system = data.getSystem();
+        Unit<Temperature> unit = "F".equals(system.getUnit()) ? ImperialUnits.FAHRENHEIT : SIUnits.CELSIUS;
+
+        if (channelUID.getId().startsWith(CHANNEL_PREFIX)) {
+            int channelId = Integer.parseInt(groupId.substring(CHANNEL_PREFIX.length())) - 1;
+            if (channelList.size() > 0 && channelId < channelList.size()) {
+                Channel channel = channelList.get(channelId);
+                switch (channelUID.getIdWithoutGroup()) {
+                    case CHANNEL_NAME:
+                        if (command instanceof StringType) {
+                            channel.setName(command.toFullString());
+                            return true;
+                        }
+                    case CHANNEL_MIN:
```

Comment:
Is this fall through intentional?

Reply:
No, this bug was introduced by the change of the return handling. Thanks, fixed.